tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -1213,23 +1265,37 @@ bool FindMatMulBiasAddAndGelu(RemapperContext* ctx, int node_index,
   // ops), we check if (i) MatMul op is CpuCompatible or GpuComptible, (ii)
   // const nodes have desired values.
   if (found_gelu_exact) {
-    if (!IsMKLEnabled()) return false;
-    // Check if the MatMul to be fused is CPU compatible
-    // TODO(kaixih@nvidia): Add GPU support when cublastLt supports the exact
-    // form.
+    // Check if the MatMul to be fused is device compatible.
     NodeDef* matmul_node =
         ctx->graph_view.GetNode(matched_nodes_map->at("matmul"))->node();
-    if (!IsCpuCompatibleMatMul(*ctx, matmul_node)) {
-      matched_nodes_map->clear();
-      remove_node_indices->clear();
-      return false;
+    DataType matmul_dtype = GetDataTypeFromAttr(*matmul_node, "T");
+
+    bool cpu_ok = IsMKLEnabled() && IsCpuCompatibleMatMul(*ctx, matmul_node);
+    bool gpu_ok = NodeIsOnGpu(matmul_node) && RuntimeFusionEnabled(cluster) &&
+                  matmul_dtype == DT_HALF;
+    if (!cpu_ok && !gpu_ok) return false;
+
+    // Check if the leading dims of input matrices are even numbers. The CuDNN
+    // runtime fusion kernels require 32-bit aligned data loading.
+    if (gpu_ok) {
+      const std::vector<OpInfo::TensorProperties>& input_props =
+          ctx->graph_properties.GetInputProperties(matmul_node->name());
+      const TensorShapeProto& a_shape =
+          input_props.size() >= 1 ? input_props[0].shape() : TensorShapeProto();
+      const TensorShapeProto& b_shape =
+          input_props.size() >= 1 ? input_props[1].shape() : TensorShapeProto();
+      bool valid_dims = Rank(a_shape) == 2 && Rank(b_shape) == 2 &&
+                        IsKnown(a_shape.dim(1)) &&         //
+                        IsKnown(b_shape.dim(1)) &&         //
+                        a_shape.dim(1).size() % 2 == 0 &&  //
```

Comment:
Stray comments?

Reply:
No. I think this is a trick to avoid the clang formatter to change the alignment of these many expressions.